execution_engine/src/storage/global_state/scratch.rs

```diff
@@ -196,12 +196,45 @@ impl CommitProvider for ScratchGlobalState {
             let value = match (cached_value, transform) {
                 (None, Transform::Write(new_value)) => new_value,
                 (None, transform) => {
-                    error!(
-                        ?key,
-                        ?transform,
-                        "Key not found while attempting to apply transform"
-                    );
-                    return Err(CommitError::KeyNotFound(key).into());
+                    // It might be the case that for `Add*` operations we don't have the previous value in cache yet.
+                    let txn = self.environment.create_read_txn()?;
+                    let updated_value = match read::<
+                        Key,
+                        StoredValue,
+                        lmdb::RoTransaction,
+                        LmdbTrieStore,
+                        Self::Error,
+                    >(
+                        correlation_id,
+                        &txn,
+                        self.trie_store.deref(),
+                        &state_hash,
+                        &key,
+                    )? {
+                        ReadResult::Found(current_value) => {
+                            match transform.apply(current_value.clone()) {
+                                Ok(updated_value) => updated_value,
+                                Err(err) => {
+                                    error!(?key, ?err, "Key found, but could not apply transform");
+                                    return Err(CommitError::TransformError(err).into());
+                                }
+                            }
+                        }
+                        ReadResult::NotFound => {
+                            error!(
+                                ?key,
+                                ?transform,
+                                "Key not found while attempting to apply transform"
+                            );
+                            return Err(CommitError::KeyNotFound(key).into());
+                        }
+                        ReadResult::RootNotFound => {
+                            error!(root_hash=?state_hash, "root not found");
+                            return Err(CommitError::ReadRootNotFound(state_hash).into());
+                        }
+                    };
+                    txn.commit()?;
+                    updated_value
```

Comment:
I copied (with slight modifications) this piece from another place. Maybe there's an option to avoid that but I didn't think about it when I was testing the quick-fix.